pubsub.go

```diff
@@ -418,6 +427,8 @@ func (p *PubSub) handleRemoveSubscription(sub *Subscription) {
 
 	sub.err = fmt.Errorf("subscription cancelled by calling sub.Cancel()")
 	close(sub.ch)
+	close(sub.joinCh)
+	close(sub.leaveCh)
```

Comment:
let's move these three closures into a `close` method in the subscription.

Reply:
ok, private method right?